project.go

```diff
@@ -43,12 +43,32 @@ func findProjectRoot(from string) (string, error) {
 type Project struct {
 	// AbsRoot is the absolute path to the root directory of the project.
 	AbsRoot string
+	// ResolvedAbsRoot is the resolved absolute path to the root directory of the project.
+	// If AbsRoot is not a symlink, then ResolvedAbsRoot should equal AbsRoot.
+	ResolvedAbsRoot string
 	// ImportRoot is the import path of the project's root directory.
 	ImportRoot gps.ProjectRoot
 	Manifest   *Manifest
 	Lock       *Lock // Optional
 }
 
+// SetRoot set the project AbsRoot. If the root is a symlink, it attempts to resolve it.
+func (p *Project) SetRoot(root string) error {
+	p.AbsRoot = root
+	p.ResolvedAbsRoot = root
```

Comment:
It's probably preferable that we _only_ change the `*Project`'s internal state if `filepath.EvalSymlinks()` exits without an error. This makes it generally consistent with the practice that Go functions should not return values from funcs when an error is also returned - just, in this case, those values are internal properties, rather than being return values.

Does that make sense?